test/signalrclient-e2e-tests/connection_tests.cpp

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#pragma once
+
+#include "stdafx.h"
+#include <string>
+#include "cpprest\details\basic_types.h"
+#include "cpprest\json.h" 
+#include "connection.h"
+#include "hub_connection.h"
+#include "console_log_writer.h"
+
+using namespace signalr;
+using namespace utility;
+using namespace std;
+
+TEST(connection_tests, send_message)
+{
+	shared_ptr<log_writer> Log_writer(make_shared<console_log_writer>());
+
+	connection conn(L"http://localhost:8081/raw-connection", L"", trace_level::all, Log_writer);
+
+	string_t message;
+	int count = 0;
+	pplx::event test_event{};
+
+	conn.set_message_received([&message, &count, &test_event](const string_t& payload)
+	{
+		message = payload;
+		wcout << "Received message: " << payload << endl;
+
+		count += 1;
+		if (count == 2)
+		{
+			test_event.set();
+		}
+	});
+
+	conn.start().then([&conn](pplx::task<void> start_task)
+	{
+		start_task.get();
+		ASSERT_TRUE(conn.get_connection_state() == connection_state::connected);
+
+		web::json::value obj;
+		obj[L"type"] = web::json::value::number(1);
+		obj[L"value"] = web::json::value::string(L"test");
+
+		conn.send(obj.serialize())
```

Comment:
this probably does not matter here because you are using an event to block but I already burnt myself by not returning a task for pplx::task<void> continuations (note - this is fire and forget here)